truss/templates/server/model_wrapper.py

```diff
@@ -430,17 +434,33 @@ def _load_impl(self):
 
         self._maybe_model_descriptor = ModelDescriptor.from_model(self._model)
 
+    async def try_load(self):
+        await to_thread.run_sync(self._initialize_model)
+
         if self._maybe_model_descriptor.setup_environment:
             self._initialize_environment_before_load()
 
         if hasattr(self._model, "load"):
-            retry(
-                self._model.load,
-                NUM_LOAD_RETRIES,
-                self._logger.warning,
-                "Failed to load model.",
-                gap_seconds=1.0,
-            )
+            if inspect.iscoroutinefunction(self._model.load):
+                async for attempt in AsyncRetrying(
+                    stop=stop_after_attempt(NUM_LOAD_RETRIES),
+                    wait=wait_fixed(1),
+                    before_sleep=lambda retry_state: self._logger.info(
+                        f"Model load failed (attempt {retry_state.attempt_number})...retrying"
+                    ),
+                ):
+                    with attempt:
+                        (await self._model.load(),)
+
+            else:
+                await to_thread.run_sync(
+                    retry,
```

Comment:
Could you kill this "custom" `retry` and also use tenacity here?

Reply:
yeah for sure